File: my-folder/0145-binary-tree-postorder-traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
#class Solution:
    #def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
     #   ret = []
     #   if root is None:
     #       return

       # ret += self.postorderTraversal(root.left)
       # ret += self.postorderTraversal(root.right)
      #  ret.append(root.val)
       # return ret#

class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ret = []
        if root is None:
            return ret

        ret += self.postorderTraversal(root.left)
        ret += self.postorderTraversal(root.right)
        ret.append(root.val)
        return ret

        

File: my-folder/0145-binary-tree-postorder-traversal/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int>ret;
    void post(TreeNode* root) {
        
        if(root->left)  post(root->left);
        if(root->right)  post(root->right);
        ret.push_back(root->val);
    }
    vector<int> postorderTraversal(TreeNode* root) {
        
        if(root == NULL) return ret;
        post(root);
        return ret;
    }
};